### PR TITLE
Allow duplicate marking in the germline subworkflow in umccrise

### PR DESCRIPTION
**Pull request by alexiswl**

### CODE REVIEW

File: workflows/umccrise-with-dragen-germline-pipeline/2.1.1--3.9.3/umccrise-with-dragen-germline-pipeline__2.1.1--3.9.3.cwl [hunk 1]
```diff
@@ -94,6 +94,11 @@ inputs:
       The name of the output directory used for umccrise step
     type: string
   # Optional germline inputs
+  enable_duplicate_marking_germline:
+    label: enable duplicate marking germline
+    doc: |
+      Turn on duplicate marking for the germline step
+    type: boolean?
```
Comment (skanwal) [inline workflows/umccrise-with-dragen-germline-pipeline/2.1.1--3.9.3/umccrise-with-dragen-germline-pipeline__2.1.1--3.9.3.cwl:101]: For convenience, should we set default to true? As in almost all cases, we are going to use this argument anyway.

Comment (alexiswl) [reply workflows/umccrise-with-dragen-germline-pipeline/2.1.1--3.9.3/umccrise-with-dragen-germline-pipeline__2.1.1--3.9.3.cwl:101]: On the fence on this one. 

* Yes purely for the fact that we will always set this and would have saved us a lot of trouble. 
* No we don't set it as a default for any other pipelines, and so should be set in the run json explicitly. The workflow should try and represent the default behaviour of the tool, which is false.

Comment (skanwal) [reply workflows/umccrise-with-dragen-germline-pipeline/2.1.1--3.9.3/umccrise-with-dragen-germline-pipeline__2.1.1--3.9.3.cwl:101]: > No we don't set it as a default for any other pipelines

We don't set as default but then set the step input to `true` https://github.com/umccr/cwl-ica/blob/main/workflows/dragen-wgs-qc-pipeline/3.9.3/dragen-wgs-qc-pipeline__3.9.3.cwl#L100. That's why wanted to mimic same behaviour here.

Comment (ohofmann) [reply workflows/umccrise-with-dragen-germline-pipeline/2.1.1--3.9.3/umccrise-with-dragen-germline-pipeline__2.1.1--3.9.3.cwl:101]: I'm also not sure about the _default_ state but is there a way to make a subset of DRAGEN command line options (or that of any command, really) mandatory?

We don't have to worry about 90% of DRAGEN's runtime options but there are a few where we should be explicit and require the workflow to make a clear choice of how it uses these parameters.

Comment (alexiswl) [reply workflows/umccrise-with-dragen-germline-pipeline/2.1.1--3.9.3/umccrise-with-dragen-germline-pipeline__2.1.1--3.9.3.cwl:101]: > I'm also not sure about the default state but is there a way to make a subset of DRAGEN command line options (or that of any command, really) mandatory?

Yes we could set this as a required boolean so the user is forced to specify either true or false or the pipeline will fail

Comment (alexiswl) [reply workflows/umccrise-with-dragen-germline-pipeline/2.1.1--3.9.3/umccrise-with-dragen-germline-pipeline__2.1.1--3.9.3.cwl:101]: Will close this PR and create a new one with required booleans update for both germline and somatic.

**Pull request closed without merging**